app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -68,6 +75,7 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
                 nino ← EitherT.fromOption[Future](maybeNino, "could not retrieve either userDetailsUrl or NINO from auth")
                 eligible ← helpToSaveService.checkEligibility(nino, authorisationCode)
                 nsiUserInfo ← toNSIUserInfo(eligible)
+                _ <- EitherT.fromEither[Future](validateCreateAccountJsonSchema(nsiUserInfo))
```

Comment:
why do we have two validations in place now ,  one inside of `toNSIUserInfo(eligible)` and one in `validateCreateAccountJsonSchema` ?